Source/NSHost.m

```diff
@@ -530,6 +530,19 @@ + (NSHost*) hostWithAddress: (NSString*)address
 #if     defined(AF_INET6)
     {
       struct in6_addr	hostaddr6;
+      
+      //Remove brackets if exists
+      const char *formattedAddress;
+      NSUInteger addressLength = [address length] - 1; //Don't fall off the array
+      char firstChar = [address characterAtIndex:0];
+      char lastChar = [address characterAtIndex:addressLength];
+
+      if(firstChar == '[' && lastChar == ']')
+      {
+          NSString *lastBracketRemoved = [address substringToIndex:addressLength];
+          NSString *cleanedAddress = [lastBracketRemoved substringFromIndex:1];
+          a = [cleanedAddress UTF8String];
```

Comment:
is there not a much easier way to do this? something like [NSString substringWithRange]?

Reply:
That approach could work, I would still need to find the length of the address and create a range from that. The reason being, we won't know what format the IPv6 address will be in.

Reply:
Its the same I think. I will close this thread.